service/telemetry/telemetry.go

```diff
@@ -91,3 +115,63 @@ func toSamplingConfig(sc *LogsSamplingConfig) *zap.SamplingConfig {
 		Thereafter: sc.Thereafter,
 	}
 }
+
+func getRotationSinkFactory(cfg *configrotate.Config) func(u *url.URL) (zap.Sink, error) {
+	return func(u *url.URL) (zap.Sink, error) {
+		p := u.Query().Get("path")
+		writer, err := cfg.NewWriter(p)
+		if err != nil {
+			return nil, err
+		}
+		return nopSyncSink{writer}, nil
+	}
+}
+
+// lumberjack.Logger does not provide a Sync() method, which is required by zap.Sink
+// explanation: https://github.com/natefinch/lumberjack/pull/47#issuecomment-322502210
+type nopSyncSink struct {
+	io.WriteCloser
+}
+
+func (w nopSyncSink) Sync() error {
+	return nil
+}
+
+func setRotatinURL(paths []string, rotationSchema string) ([]string, error) {
+	res := make([]string, 0, len(paths))
+	for _, p := range paths {
+		if runtime.GOOS == "windows" && filepath.IsAbs(p) {
+			res = append(res, rotationSchema+":?path="+url.QueryEscape(p))
+			continue
+		}
+		u, err := url.Parse(p)
+		if err != nil {
+			return nil, err
+		}
+		if (u.Scheme == "" || u.Scheme == "file") &&
+			u.Path != "stdout" && u.Path != "stderr" {
+			// Copied from zap. Only clean URLs are allowed
+			if u.User != nil {
+				return nil, fmt.Errorf("user and password not allowed with file URLs: got %v", u)
+			}
+			if u.Fragment != "" {
+				return nil, fmt.Errorf("fragments not allowed with file URLs: got %v", u)
+			}
+			if u.RawQuery != "" {
+				return nil, fmt.Errorf("query parameters not allowed with file URLs: got %v", u)
+			}
+			// Error messages are better if we check hostname and port separately.
+			if u.Port() != "" {
+				return nil, fmt.Errorf("ports not allowed with file URLs: got %v", u)
+			}
+			if hn := u.Hostname(); hn != "" && hn != "localhost" {
+				return nil, fmt.Errorf("file URLs must leave host empty or use localhost: got %v", u)
+			}
+
+			res = append(res, rotationSchema+":?path="+url.QueryEscape(u.Path))
+			continue
+		}
+		res = append(res, p)
```

Comment:
what happens if the scheme is http?

Reply:
Zap allows us to register schemata. By default, only `file` is registered. Unregistered schemata will cause errors.